### ISSUE TITLE
External redirect

### ISSUE THREAD

**Issue by Pako2** [fullphat/snarl#25]
Hi !
I would like to update my EG_SnarlRedirect for version 5.0.
The current version not only does not work but it can not even be installed.
How do I register an external redirect for version 5.0, when no snarlm.exe exists?

**Comment by fullphat** [comment]
Hi Pako! 😀 

Glad you want to upgrade EG_SnarlRedirect!  At the moment R5.0 doesn't support any sort of external redirect, but we will be adding that functionality in Beta 3.  So far as EG_SnarlRedirect goes, you have two options:

1. (Preferred) recode the redirect to .net.  Benefits here include much better integration with Snarl R5.0, much easier installation (just drop the DLL into Snarl's %APPDATA% folder) plus you get access to all the .net features.

2. Leave your extension as-is - we'll include support for legacy (COM) redirects soon.

If you fancy option (1), let me know so we can work on the Redirect API together!

**Comment by Pako2** [comment]
Hi Chris !
Thank you for answer.
I also prefer option 1.
Can I see some functional example (include source code)?
By the way - you can see my current program [here](https://sourceforge.net/p/snarlegredirect/code/HEAD/tree/trunk/).

Thanks,
Pako

**Comment by fullphat** [comment]
Awesome!  No API yet, but how about the following as a start?

https://github.com/fullphat/snarl/wiki/Services-API

**Comment by Pako2** [comment]
Thank you for the answer, but I do not know how to start from scratch.
I rarely used Visual Studio and it's been a long time, when it was the last time. Now I have a different notebook and I have to install Visual Studio 2017. This is a completely new environment for me. I do not even know (with certainty) what type of project I have to choose when I'm starting a new project ...

I've looked at the link provided, but that's not enough for me.
For example, the first sentence: "Services provide a class that implements fullphat.snarl.Service".
Where do I get "fullphat.snarl.Service"?

Mostly I proceed in the way that I study some functional example and then I will adjust it for my needs. If I can not proceed this way now, it will be very difficult for me.

**Comment by fullphat** [comment]
No, problem.  Firstly - this API doesn't exist - yet.  That can be resolved easily however. :grinning:   As for a development environment, we use Visual Studio Express 2010 right now and Snarl is built against .net 4.0.

How about we put together the library and a sample service and you take a look at this?

**Comment by Pako2** [comment]
>
> How about we put together the library and a sample service and you take a
> look at this?


Yes, of course it would be great.

I have noticed that Pushbullet redirect is now part of 5.0. That's great.
BTW - You know that I created (two years ago)
Both Snarl-PushBullet Redirect
<https://sourceforge.net/projects/snarlpbredirect/> and Snarl-Pushbullet
Extension <https://sourceforge.net/projects/snarlpbextension/>?

2017-04-18 19:53 GMT+02:00 fullphat <notifications@github.com>:

**Comment by fullphat** [comment]
Yes!  I used it a lot and it inspired me to get other web services working.  .net makes this sort of thing an absolute breeze to implement.

**Comment by Reonekot** [comment]
Sorry for thread hijack, but why the f... are you using such an old version of VS? The community editions of vs2015 and 2017 has support for plugins, which the old Express editions didn't,
 besides, imo, being vast improvement on vs2010, which is by far the worst VS edition.

**Comment by Pako2** [comment]
>
> Yes! I used it a lot and it inspired me to get other web services working.
> .net makes this sort of thing an absolute breeze to implement.


I am very pleased that my work was not unnecessary.
Only Pushbullet redirect is currently integrated. Are you also planning to
integrate the Pushbullet extension (that is, Snarl behaves like a
Pushbullet device)?

2017-04-19 18:33 GMT+02:00 fullphat <notifications@github.com>:

**Comment by fullphat** [comment]
No immediate plans, but it is something we'd like to see down the road one we've sorted out the extension API.  At the moment, the approximate running order is:

Beta 2: networking baseline sorted
Beta 3: services (aka redirects) API in place
Beta 4: SnarlOSD improvements
Beta 5: extensions API in place
Beta 6: sysadmin features in place

Of course, this is completely indicative and we may alter the running order if there's a clear demand for it.
